client/state/timezones/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { timezonesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Collect and store timezones data
+ *
+ * @param  {Object} state - Current state
+ * @param  {Object} action - action payload
+ * @return {Object} updated state
+ */
+export const items = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, {
+		rawOffsets,
+		timezonesByContinent,
+	} ) => ( { rawOffsets, timezonesByContinent } ),
+}, timezonesSchema );
+
+/**
+ * Track the timezones requesting process
```

Comment:
> Since the HTTP layer has been merged (#10327) could I just remove the isRequesting reducer?

Sounds like a great plan. We're going to make sure that each individual API endpoint handler need not worry about this trivial case of "make sure I don't send out a request while one is still in transit"